### ISSUE TITLE
[BUG] `openqasm3_to_ionq` pyqasm qubit registers empty

### ISSUE THREAD

**Issue by ryanhill1** [qBraid/qBraid#864]
The following lines were missing code coverage after the recent `pyqasm` integration:

https://github.com/qBraid/qBraid/blob/59524227a1026de988ea0890aed8fe02b9ebf926/qbraid/transpiler/conversions/openqasm3/openqasm3_to_ionq.py#L153-L158

The following test case, if added to `tests/transpiler/qasm/test_qasm_to_ionq.py`, would account for this gap:

```python
from qbraid.transpiler.conversions.openqasm3.openqasm3_to_ionq import openqasm3_to_ionq

def test_openqasm3_to_ionq_deutch_jozsa(deutsch_jozsa_qasm3, deutch_jozsa_ionq):
    """Test transpiling the Deutsch-Jozsa algorithm from openqasm3.ast.Program to IonQDictType."""
    program = parse(deutsch_jozsa_qasm3)
    ionq_program = openqasm3_to_ionq(program)
    assert ionq_program == deutch_jozsa_ionq
```

However, the test case fails because `program_qubits` (below) is empty for all cases.

https://github.com/qBraid/qBraid/blob/59524227a1026de988ea0890aed8fe02b9ebf926/qbraid/transpiler/conversions/openqasm3/openqasm3_to_ionq.py#L142

It seems that `pyqasm.QasmModule._qubit_registers` only gets initialized after some other operation e.g. `unroll()`, `validate()`, `num_qubits`, `reverse_qubit_order()`, etc. For this test to pass, we need to somehow populate the `program.module._qubit_registers` attribute. However, calling any one of the above methods to do so raises various exceptions in the other QASM to IonQ tests.

**Comment by TheGupta2012** [comment]
I think the best way is to modify the implementation of  populating `_qubit_registers` on `pyqasm` side

**Comment by TheGupta2012** [comment]
Another thing to note here is that many of the validations like [`IndexError` for parameter validations](https://github.com/qBraid/qBraid/blob/59524227a1026de988ea0890aed8fe02b9ebf926/qbraid/transpiler/conversions/openqasm3/openqasm3_to_ionq.py#L170) and [`ValueError` for argument validations](https://github.com/qBraid/qBraid/blob/59524227a1026de988ea0890aed8fe02b9ebf926/qbraid/transpiler/conversions/openqasm3/openqasm3_to_ionq.py#L235) are already enforced in `pyqasm` definitions. These cases are indeed caught in `pyqasm` but the caveat is that the context related to IONQ conversion is not present in the error messages. We can remove these extra validations from the converter and catch the errors raised by `pyqasm` but I feel current approach is better so that we can have more granularity about the error messages to be raised during conversions

**Comment by ryanhill1** [comment]
Yeah I agree taking the pyqasm validation out for now and sticking with the manual exception handling is a better user experience. Because the pyqasm error messages still need a good amount of work. A lot of them are still uncaught `TypeError`s that are a bit cryptic.
